crates/anvil-polkadot/tests/it/utils.rs

```diff
@@ -124,6 +130,28 @@ impl TestNode {
         StorageKey(key)
     }
 
+    /// Execute an ethereum transaction.
+    pub async fn send_transaction(
+        &mut self,
+        transaction: TransactionRequest,
+        block_number: Option<u32>,
+    ) -> Result<H256, Box<dyn std::error::Error>> {
+        let tx_hash = unwrap_response::<H256>(
+            self.eth_rpc(EthRequest::EthSendTransaction(Box::new(WithOtherFields::new(
+                transaction,
+            ))))
+            .await
+            .unwrap(),
+        )?;
+
+        if let Some(block_nr) = block_number {
+            self.wait_for_block_with_timeout(block_nr, std::time::Duration::from_secs(5))
```

Comment:
Is there a reason we wait so much for a new block? 
nit: Also maybe it would be worth to check that we are not in a no_mining state before waiting